cmd/heartbeat/heartbeat.go

```diff
@@ -170,6 +182,23 @@ func LoadParams(v *viper.Viper) (paramscmd.Params, error) {
 	}, nil
 }
 
+// RateLimited determines if we should send heartbeats to the API or save to the offline db.
+func RateLimited(offline params.Offline) bool {
```

Comment:
Instead passing the whole struct, let's keep it simple by only passing what's used in the function. It's also missing tests for this function.
```suggestion
func RateLimited(offlineDisabled bool, offlineRateLimit int, offlineLastSentAt time) bool {
```

Reply:
I actually like passing the struct here because it's called from two different files where we have the struct already, and we don't really care about the details of the individual params, we mainly care about the return value from `RateLimited`. It makes the code read as english easier:

`if rate limited` vs `if rate limited given bool, int, time`

Reply:
Instead we could create a struct to hold this params like we've done [here](https://github.com/wakatime/wakatime-cli/blob/develop/cmd/heartbeat/heartbeat.go#L234-L273).

Reply:
Done.